### ISSUE TITLE
Feature request: Narrow LogLevel type to not allow arbitrary strings

### ISSUE THREAD

**Issue by danrivett** [aws-powertools/powertools-lambda-typescript#1309]
### Use case

Raising this ticket after the discussion in [#1198](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/1198#issuecomment-1435041684).

Currently the `LogLevel` type allows for any string to be used, but at runtime it [validates the value](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/logger/src/Logger.ts#L745) and if it doesn't match one of the valid log levels (as per `isValidLogLevel()`) it silently ignores the log level passed in and uses the default.

This has two problems:

1. The issue isn't caught at compile time
2. The passed in log level is silently ignored which may not be detected.

Regarding 1. above, the following typo currently compiles. It would be better for the typo to be detected at compile time:

```ts
const logger = new Logger({ logLevel: 'DEBG' });
```

### Solution/User Experience

Narrow the `LogLevel` type definition, removing  the `| string` part.

### Acknowledgment

- [X] This feature request meets [Lambda Powertools Tenets](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/#tenets)
- [ ] Should this be considered in other Lambda Powertools languages? i.e. [Python](https://github.com/awslabs/aws-lambda-powertools-python/), [Java](https://github.com/awslabs/aws-lambda-powertools-java/)

**Comment by dreamorosi** [comment]
Hi Dan, thank you for opening the issue.

I think this is a valuable addition to the logger, I'll be working on this.
